plugin-authentication/objecten-api-authentication/src/main/kotlin/com/ritense/objectenapiauthentication/ObjectenApiAuthenticationPlugin.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.objectenapiauthentication
+
+import com.ritense.objectenapi.ObjectenApiAuthentication
+import com.ritense.plugin.annotation.Plugin
+import com.ritense.plugin.annotation.PluginProperty
+import org.springframework.web.reactive.function.client.ClientRequest
+import org.springframework.web.reactive.function.client.ClientResponse
+import org.springframework.web.reactive.function.client.ExchangeFunction
+import reactor.core.publisher.Mono
+
+@Plugin(
+    key = "objectenapiauthentication",
+    title = "Objecten API Authentication",
+    description = "Plugin used to provide authentication for Objecten API"
+)
+class ObjectenApiAuthenticationPlugin : ObjectenApiAuthentication {
```

Comment:
I feel like this could be more generic, like 'PreSharedTokenAuthentication'? Let's discuss.

Reply:
I wouldn't be against it. Just not to sure if it is obvious for frontend users that they should create that plugin before they can use the Objecten API and Objecttypen API